src/neuron.js

```diff
@@ -46,16 +46,21 @@ Neuron.prototype = {
 
     // old state
     this.old = this.state;
-
     // eq. 15
-    this.state = this.selfconnection.gain * this.selfconnection.weight *
-      this.state + this.bias;
+    this.state = this.selfconnection.gain * this.selfconnection.weight * this.state + this.bias;
 
     for (var i in this.connections.inputs) {
       var input = this.connections.inputs[i];
-      this.state += input.from.activation * input.weight * input.gain;
+      if(input.gater){
```

Comment:
I don't see what you are achieving here, if an input is a not a gater, the `gain` is always `1`, so there's no need for an if statement here and the second for loop below on line 58